src/System.IO.FileSystem/src/System/IO/DirectoryInfo.cs

```diff
@@ -56,14 +56,16 @@ public DirectoryInfo CreateSubdirectory(string path)
         {
             if (path == null)
                 throw new ArgumentNullException(nameof(path));
-            if (path.Length == 0)
+            if (PathInternal.IsEffectivelyEmpty(path))
                 throw new ArgumentException(SR.Argument_PathEmpty, nameof(path));
             if (Path.IsPathRooted(path))
                 throw new ArgumentException(SR.Arg_Path2IsRooted, nameof(path));
 
             string fullPath = Path.GetFullPath(Path.Combine(FullPath, path));
 
-            if (0 != string.Compare(FullPath, 0, fullPath, 0, FullPath.Length, PathInternal.StringComparison))
+            if (0 != string.Compare(FullPath, 0, fullPath, 0, FullPath.Length, PathInternal.StringComparison) 
```

Comment:
If yo'ure changing this you can fix the [Yoda conditional](https://en.wikipedia.org/wiki/Yoda_conditions). Better still, change to string.Equals instead. It can be faster (eg if strings are different lengths it bails immediately)

Reply:
String.Equals does not have any overload to match a string with a substring of second string.

Reply:
I missed that, you're right. Maybe there should be a STring.Equals(Span, Span)...